syntax = "proto3";
package container;




import "model/topology/container.proto";

/*
ListContainer请求
*/
message ListContainerRequest {
    /*
    页码
    */
    int32 page = 1 ;
    /*
    每页数据量
    */
    int32 pageSize = 2 ;
}

/*
ListContainer返回
*/
message ListContainerResponse {
    /*
    页数
    */
    int32 page = 1 ;
    /*
    页大小
    */
    int32 page_size = 2 ;
    /*
    总数
    */
    int32 total = 3 ;
    /*
    数据列表
    */
    repeated topology.Container list = 4 ;
}

/*
ListContainerApi返回
*/
message ListContainerResponseWrapper {
    /*
    返回码
    */
    int32 code = 1 ;
    /*
    返回码解释
    */
    string codeExplain = 2 ;
    /*
    错误详情
    */
    string error = 3 ;
    /*
    返回数据
    */
    container.ListContainerResponse data = 4 ;
}

